e2etest/pkg/ipfamily/ipfamily.go

```diff
@@ -39,7 +40,7 @@ func ForAddresses(ips []string) (Family, error) {
 			return Unknown, fmt.Errorf("IPFamilyForAddresses: Invalid address %q", ips)
 		}
 		if (ip1.To4() == nil) == (ip2.To4() == nil) {
-			return Unknown, fmt.Errorf("IPFamilyForAddresses: same address family %q", ips)
+			return PreferDualStack, nil
```

Comment:
this if means we have a pair of addresses of hte same family, why this should be preferdualstack?
Also again, I think here we are mixing the ipfamily as "what type these addresses are" and what the service would like to have.